public/app/features/scenes/dashboard/LibraryVizPanel.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+
+import { SceneComponentProps, SceneObjectBase, SceneObjectState, VizPanel } from '@grafana/scenes';
+import { LibraryPanel, LibraryPanelRef } from '@grafana/schema';
+import { PanelModel } from 'app/features/dashboard/state';
+import { getLibraryPanel } from 'app/features/library-panels/state/api';
+
+import { createPanelDataProvider } from './utils/createPanelDataProvider';
+
+interface LibraryVizPanelState extends SceneObjectState {
+  title: string;
+  libraryPanel: LibraryPanelRef | LibraryPanel;
```

Comment:
I think we can type this as only LibraryPanelRef, the typing of PanelModel.library panel as `libraryPanel?: LibraryPanelRef | LibraryPanel;`  is a huge source for complexity, when we load the dashboard initially it's always LibraryPanelRef. 

We could potentially not have this libraryPanelRef here at all but just have name + uid instead, think that could be cleanest, hm... but then we would have name and title which is not so clean, maybe something like this? 

```ts
   body: new LibraryVizPanel({
         title: panel.title, 
         ref: { uid: panel.libraryPanel.uid, name: panel.libraryPanel.name }
        }),
```

Reply:
Was going to ask about this union type and the "why" of it. But true, think this is the way to go:

```ts
interface LibraryVizPanelState extends SceneObjectState {
  title: string;
  ref: LibraryPanelRef;
}
```